src/components/ManageIdentitiesList.tsx

```diff
@@ -0,0 +1,400 @@
+import {
+  Box,
+  Button,
+  ButtonGroup,
+  Flex,
+  Text,
+  Input,
+  Alert,
+  AlertIcon,
+  AlertTitle,
+  AlertDescription,
+} from "@chakra-ui/react";
+import { useContext, useState, useEffect } from "react";
+import { useMutation } from "@tanstack/react-query";
+import { ethers } from "ethers";
+import { TokenNameItem, UpdateCountryProps } from "@/types/types";
+import { Eip3643Context } from "@/contexts/Eip3643Context";
+import { MenuSelect } from "@/components/MenuSelect";
+import {
+  readTokenName,
+  readTokenOwner,
+  writeIdentityRegistryDeleteIdentity,
+  writeIdentityRegistryUpdateCountry,
+  writeIdentityRegistryUpdateIdentity,
+  writeIdentityRegistryAddAgent,
+  watchIdentityRegistryAgentAddedEvent,
+  watchIdentityRegistryAgentRemovedEvent,
+  writeIdentityRegistryRemoveAgent,
+} from "@/services/contracts/wagmiGenActions";
+import { useTokenIdentityRegistry } from "@/hooks/useTokenIdentityRegistry";
+import { useWalletInterface } from "@/services/wallets/useWalletInterface";
+import { WalletInterface } from "@/services/wallets/walletInterface";
+import { WatchContractEventReturnType } from "@/services/contracts/watchContractEvent";
+
+const investorCountriesItems = [
+  {
+    value: 840,
+    label: "US",
+  },
+  {
+    value: 0,
+    label: "Non US",
+  },
+];
+
+export function ManageTokensList() {
+  const [addedAgents, setAddedAgents] = useState([] as Array<string>);
+  const [ownTokens, setOwnTokens] = useState([] as Array<TokenNameItem>);
+  const [updateTxResult, setUpdateTxResult] = useState<string>();
+  const [updateTxError, setUpdateTxError] = useState<string>();
+  const { deployedTokens } = useContext(Eip3643Context);
+  const { accountEvm } = useWalletInterface();
+  const [selectedToken, setSelectedToken] = useState<TokenNameItem | null>(
```

Comment:
I'd suggest dropping the null from the typing and as default, keep only TokenNameItem and the default absent undefined would be inferred i believe

Reply:
<img width="510" alt="image" src="https://github.com/user-attachments/assets/d67d0da2-e68c-4048-8fac-69e0bacc3e78">

Reply:
```
  const [selectedToken, setSelectedToken] = useState<TokenNameItem>();
```